pages/bookmarks.jsx

```diff
@@ -14,57 +13,45 @@ export default React.createClass({
       bookmarks: null
     };
   },
-  verifyLoggedInStatus() {
-    Service.userstatus()
-      .then(response => this.updateLoggedInInfo(response.loggedin, response.username))
-      .catch(reason => {
-        this.updateLoggedInInfo(false);
-        console.error(reason);
-      });
-  },
-  updateLoggedInInfo(status, username = ``) {
-    UserData.userLoggedInStatus.set(status);
-    UserData.username.set(username);
-    this.forceUpdate();
-  },
-  logOutUser(event) {
+  handleLogOutBtnClick(event) {
     event.preventDefault();
-    Service.logout()
-      .then(() => {
-        // we have successfully logged user out
-        this.updateLoggedInInfo(false);
-      })
-      .catch(reason => {
-        console.error(reason);
-      });
+    Login.logoutUser(error => {
+      if (!error) {
+        this.forceUpdate();
+      }
+    });
   },
   componentDidMount() {
-    let query = this.props.router.location.query;
-    
-    // we don't need the "loggedin" param sent back from Pulse API to be presented at all times
-    delete query.loggedin;
-
-    browserHistory.replace({
-      pathname: this.props.router.location.pathname,
-      query: query
-    });
-
     // get IDs of user's bookmarked entries
     this.setState({bookmarks: getBookmarks()});
 
     // let's verify what the logged in status is
-    this.verifyLoggedInStatus();
-  },
-  componentWillMount() {
-    console.log(`[bookmarks componentWillMount] UserData.userLoggedInStatus.get() = `, UserData.userLoggedInStatus.get());
+    let currentLocation = this.props.router.location;
+
+    Login.verifyLoggedInStatus(currentLocation, error => {
+      if (!error) {
+        this.forceUpdate();
+      }
+    });
   },
   render() {
     let redirectUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
     let logInUrl = UserData.logInUrl.get(redirectUrl);
-    let loggedInStatus = UserData.userLoggedInStatus.get() ?
-                        <p>Hi {UserData.username.get()}! <button onClick={this.logOutUser} className="btn-link">Log out</button>.</p>
-                        : <p>Are you Mozilla Staff? <a href={logInUrl}>Sign in</a>.</p>;
-    
+    let promptToLogIn = <p>Are you Mozilla Staff? <a href={logInUrl}>Sign in</a>.</p>;
```

Comment:
again, too many types here? Let's just focus on "logged in" vs. "anonymous" for now.